composer.json

```diff
@@ -9,15 +9,16 @@
         {"name": "Steve Lacey", "email": "steve@steve.ly"}
     ],
     "require": {
-        "php": "^7.2 || ^8.0",
```

Comment:
Is it really necessary to drop PHP 7 support in order to add ORM 3 support?

Note that you did not adjust the CI: It's still configured for running tests on PHP 7.2, 7.3 and 7.4.

Reply:
Current code is not compatible with pre-8 versions due to `mixed` return type. If I set the return type correctly everywhere I think it can be compatible with earlier versions again.